src/Microsoft.PowerShell.Commands.Utility/commands/utility/group-object.cs

```diff
@@ -336,25 +341,31 @@ protected override void ProcessRecord()
         /// </summary>
         protected override void EndProcessing()
         {
+            // using OrderBy to get stable sort.
+            foreach (var entry in _orderedEntries.OrderBy(e => e, _orderByPropertyComparer))
+            {
+                DoGrouping(entry, NoElement, _groups, _tupleToGroupInfoMappingDictionary, _orderByPropertyComparer);
+            }
+
             s_tracer.WriteLine(_groups.Count);
             if (_groups.Count > 0)
             {
                 if (AsHashTable)
                 {
-                    Hashtable _table = CollectionsUtil.CreateCaseInsensitiveHashtable();
+                    Hashtable table = CollectionsUtil.CreateCaseInsensitiveHashtable();
                     try
                     {
-                        foreach (GroupInfo _grp in _groups)
+                        foreach (GroupInfo grp in _groups)
                         {
                             if (AsString)
```

Comment:
Make sense to move the cycle into the  `if (AsString)` ?

Reply:
Not sure i understand what you suggest here.

Reply:
Yes, I think that is clearer.

Reply:
I thought more about performancethough :-) From this point of view it seems `for()` better than `foreach` - less allocations. Is the _groups so much that we can get a win?